src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -51,6 +89,57 @@
    (utils/set-timeout #(utils/show-popup (i18n/label :t/transaction-failed) message) 1000)))
 
 ;;;; Handlers
+;; HANDLE QR CODE
+
+(defn qr-data->send-tx-data [{:keys [address value symbol gas gasPrice public-key from-chat?]}]
+  {:pre [(not (nil? address))]}
+  (cond-> {:to address :public-key public-key}
+    value (assoc :amount value)
+    symbol (assoc :symbol symbol)
+    gas (assoc :gas (money/bignumber gas))
+    from-chat? (assoc :from-chat? from-chat?)
+    gasPrice (assoc :gas-price (money/bignumber gasPrice))))
+
+(defn extract-qr-code-details [chain qr-uri]
+  {:pre [(keyword? chain) (string? qr-uri)]}
+  ;; i don't like fetching all tokens here
+  (let [{:keys [:wallet/all-tokens]} @re-frame.db/app-db
```

Comment:
Could you please explain why do you get `all-tokens` data from @re-frame.db/app-db directly?

Reply:
because there was strictly speaking no reason to have it through a sub, since the view didn't need to react on it's change

Reply:
this doesn't look good to use app-db directly if you need access to db, better to dispatch an event and get db from event's cofx

Reply:
@flexsurfer It must not be a triggered event. re-frame cycles don't work well with the qr scanner component, there can be race conditions. That was the cause of that weird QR error on ios. 

In any case there is probably a nicer solution, e.g. passing it as a param from one of the enclosing components.

But out of curiosity, is there any technical reason why you think this is bad, or just that it doesn't look pretty (on which we can agree)?